cmd/iam.go

```diff
@@ -819,12 +815,21 @@ func (sys *IAMSys) NewServiceAccount(ctx context.Context, parentUser string, ses
 		return auth.Credentials{}, errNoSuchUser
 	}
 
-	if cr.IsTemp() {
+	// Disallow service accounts to further create more service accounts.
+	if cr.IsServiceAccount() {
+		return auth.Credentials{}, errIAMActionNotAllowed
+	}
+
+	// FIXME: Disallow temporary users with no parent, this is most
+	// probably with OpenID which we don't support to provide
+	// any parent user, LDAPUsersType and MinIOUsersType are
+	// currently supported.
+	if cr.ParentUser == "" && cr.IsTemp() {
 		return auth.Credentials{}, errIAMActionNotAllowed
 	}
 
 	m := make(map[string]interface{})
-	m[parentClaim] = parentUser
+	m[parentClaim] = cr.ParentUser
```

Comment:
This looks it needs to be 'parentUser' instead of cr.ParentUser. Regular users do not have a parent user.

Reply:
I need an IF condition